Kernel/Graphics/VirtIOGPU/GraphicsAdapter.cpp

```diff
@@ -412,8 +417,20 @@ Graphics::VirtIOGPU::ResourceID VirtIOGraphicsAdapter::allocate_resource_id()
 
 Graphics::VirtIOGPU::ContextID VirtIOGraphicsAdapter::allocate_context_id()
 {
-    // FIXME: This should really be tracked using a bitmap, instead of an atomic counter
-    return m_context_id_counter++;
+    // Note: Context ID 0 is invalid.
+    Graphics::VirtIOGPU::ContextID new_context_id = 0;
```

Comment:
Seems like it would be better to return an ErrorOr<ConetxtID> instead of relying on magical error values. Magic / Sentinal values introduce the possibility of failure to check for error.

Reply:
It's been a while since I wrote this code, but I'm pretty sure that the context can't be 0 thing was a requirement in VirGL, not our code (libvirglrenderer started complaining in the host qemu session if you tried to create a context id with id 0).

Reply:
Maybe we should specify in the comment though.

Reply:
Ah, I think I misread your comment. You are correct, the ErrorOr would be the preferred way to do it.